### ISSUE TITLE
Lockfile out of sync

### ISSUE THREAD

**Issue by jsirex** [berkshelf/berkshelf#1501]
I almost always facing with such issue:

```
$ berks outdated
The lockfile is out of sync! Run `berks install` to sync the lockfile.
$ berks install
Using 7-zip (1.0.2)
Using apache2 (3.1.0)
Using apt (2.9.2)
Using ark (0.9.0)
Using chef-client (4.3.1)
...
Using windows (1.38.2)
# No installation happens, all cookbooks present, lock file hasn't changed
$ berks outdated
The lockfile is out of sync! Run `berks install` to sync the lockfile.
```

And I'm not sure which information I can provide to help resolve this issue.

**Comment by Annih** [comment]
Hello @jsirex 

Could you provide detailed info, running in debug mode: `berks outdated --debug` in your case.
It'll provide info on why it's out-of-sync.
We encountered similar issues, because latest version of berkshelf seems to have harder cookbook location enforcement.

**Comment by jsirex** [comment]
```
± |master {1} ✓| → berks outdated -b aws-monitoring-prod -d       
I, [2016-02-15T18:00:47.198902 #24690]  INFO -- : Checking if lockfile is trusted
D, [2016-02-15T18:00:47.199016 #24690] DEBUG -- : Checking proj-base (~> 0.5)
D, [2016-02-15T18:00:47.200111 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-apache-2.6.0/chefignore'
D, [2016-02-15T18:00:47.249355 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-apache-4.3.1/chefignore'
D, [2016-02-15T18:00:47.290662 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/collectd-lib-3.0.1/chefignore'
D, [2016-02-15T18:00:47.294677 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-platform-1.5.0/chefignore'
D, [2016-02-15T18:00:47.307718 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-rabbitmq-0.2.0/chefignore'
D, [2016-02-15T18:00:47.311245 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-mongodb-2.6.0/chefignore'
D, [2016-02-15T18:00:47.319364 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-apache-4.0.0/chefignore'
D, [2016-02-15T18:00:47.346491 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-elasticsearch-2.3.2/chefignore'
D, [2016-02-15T18:00:47.351634 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-grafana-0.3.0/chefignore'
D, [2016-02-15T18:00:47.355307 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/windows-1.36.1/chefignore'
D, [2016-02-15T18:00:47.364060 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/mongodb-2.1.0/chefignore'
D, [2016-02-15T18:00:47.371015 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-platform-2.3.2/chefignore'
D, [2016-02-15T18:00:47.384179 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-platform-2.2.0/chefignore'
D, [2016-02-15T18:00:47.392267 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-elasticsearch-2.3.0/chefignore'
D, [2016-02-15T18:00:47.406431 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-mongodb-1.3.0/chefignore'
D, [2016-02-15T18:00:47.414602 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-elasticsearch-1.2.1/chefignore'
D, [2016-02-15T18:00:47.423086 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-seyren-0.3.0/chefignore'
D, [2016-02-15T18:00:47.434816 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-apache-4.2.1/chefignore'
D, [2016-02-15T18:00:47.463611 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-docs-0.2.0/chefignore'
D, [2016-02-15T18:00:47.473481 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-production-firewall-0.2.1/chefignore'
D, [2016-02-15T18:00:47.484369 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-apache-4.3.2/chefignore'
D, [2016-02-15T18:00:47.516275 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-haproxy-0.2.0/chefignore'
D, [2016-02-15T18:00:47.519028 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-platform-1.4.1/chefignore'
D, [2016-02-15T18:00:47.538589 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/mongodb-2.1.1/chefignore'
D, [2016-02-15T18:00:47.544961 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-hazelcast-1.4.4/chefignore'
D, [2016-02-15T18:00:47.571355 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/simple-hazelcast-0.2.1/chefignore'
D, [2016-02-15T18:00:47.575162 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-mongodb-2.2.1/chefignore'
D, [2016-02-15T18:00:47.583166 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/simple-kibana-0.4.3/chefignore'
D, [2016-02-15T18:00:47.587119 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-base-1.2.0/chefignore'
D, [2016-02-15T18:00:47.603054 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-platform-2.3.3/chefignore'
D, [2016-02-15T18:00:47.617963 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-rabbitmq-0.5.0/chefignore'
D, [2016-02-15T18:00:47.623864 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-apache-2.3.1/chefignore'
D, [2016-02-15T18:00:47.643147 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-graph-explorer-0.3.0/chefignore'
D, [2016-02-15T18:00:47.647983 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/simple-mailcatcher-0.1.0/chefignore'
D, [2016-02-15T18:00:47.656154 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-grafana-0.4.0/chefignore'
D, [2016-02-15T18:00:47.675360 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-kibana-1.4.1/chefignore'
D, [2016-02-15T18:00:47.692149 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-mongodb-2.6.3/chefignore'
D, [2016-02-15T18:00:47.700542 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/simple-nexus-0.1.0/chefignore'
D, [2016-02-15T18:00:47.710304 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-collectd-1.4.3/chefignore'
D, [2016-02-15T18:00:47.732345 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/simple-hazelcast-1.0.0/chefignore'
D, [2016-02-15T18:00:47.736162 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-rabbitmq-0.7.3/chefignore'
D, [2016-02-15T18:00:47.742239 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-apache-4.3.0/chefignore'
D, [2016-02-15T18:00:47.771808 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-hazelcast-1.2.3/chefignore'
D, [2016-02-15T18:00:47.783128 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/rsyslog-2.1.0/chefignore'
D, [2016-02-15T18:00:47.788860 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/graphite-1.0.4/chefignore'
D, [2016-02-15T18:00:47.807932 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-docs-0.3.0/chefignore'
D, [2016-02-15T18:00:47.811287 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/simple-logstash-0.2.4/chefignore'
D, [2016-02-15T18:00:47.815652 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-elasticsearch-2.3.3/chefignore'
D, [2016-02-15T18:00:47.826272 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-logstash-queue-1.4.2/chefignore'
D, [2016-02-15T18:00:47.845322 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-firewall-0.2.0/chefignore'
D, [2016-02-15T18:00:47.851083 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/simple-kibana-0.4.1/chefignore'
D, [2016-02-15T18:00:47.868538 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-logstash-3.3.0/chefignore'
D, [2016-02-15T18:00:47.883593 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-platform-2.1.1/chefignore'
D, [2016-02-15T18:00:47.917894 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/simple-logstash-0.2.2/chefignore'
D, [2016-02-15T18:00:47.941844 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-seyren-0.5.0/chefignore'
D, [2016-02-15T18:00:47.946991 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-kibana-storage-1.3.1/chefignore'
D, [2016-02-15T18:00:47.958767 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-logstash-queue-1.1.0/chefignore'
D, [2016-02-15T18:00:47.966067 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-elasticsearch-1.2.3/chefignore'
D, [2016-02-15T18:00:47.992199 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-hazelcast-1.6.0/chefignore'
D, [2016-02-15T18:00:48.001976 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-logstash-3.1.0/chefignore'
D, [2016-02-15T18:00:48.011320 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-java-2.1.0/chefignore'
D, [2016-02-15T18:00:48.031152 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-kibana-1.2.0/chefignore'
D, [2016-02-15T18:00:48.039737 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-graph-explorer-0.2.0/chefignore'
D, [2016-02-15T18:00:48.067056 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/collectd-2.1.2/chefignore'
D, [2016-02-15T18:00:48.086633 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-graphite-0.5.0/chefignore'
D, [2016-02-15T18:00:48.091967 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-elasticsearch-1.2.4/chefignore'
D, [2016-02-15T18:00:48.096545 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-collectd-1.4.0/chefignore'
D, [2016-02-15T18:00:48.100333 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-hazelcast-1.3.1/chefignore'
D, [2016-02-15T18:00:48.105290 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-base-1.2.1/chefignore'
D, [2016-02-15T18:00:48.111292 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-collectd-1.5.0/chefignore'
D, [2016-02-15T18:00:48.125760 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-nexus-0.4.0/chefignore'
D, [2016-02-15T18:00:48.129339 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-chef-0.1.0/chefignore'
D, [2016-02-15T18:00:48.141266 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-logstash-3.2.0/chefignore'
D, [2016-02-15T18:00:48.146952 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-base-1.1.0/chefignore'
D, [2016-02-15T18:00:48.151485 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-base-0.5.1/chefignore'
D, [2016-02-15T18:00:48.155621 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-kibana-1.4.0/chefignore'
D, [2016-02-15T18:00:48.159696 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-logstash-1.6.1/chefignore'
D, [2016-02-15T18:00:48.166774 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-platform-1.7.0/chefignore'
D, [2016-02-15T18:00:48.174214 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-graphite-0.3.1/chefignore'
D, [2016-02-15T18:00:48.186026 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-chef-0.2.2/chefignore'
D, [2016-02-15T18:00:48.194427 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-production-firewall-0.3.0/chefignore'
D, [2016-02-15T18:00:48.201977 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/packagecloud-0.0.17/chefignore'
D, [2016-02-15T18:00:48.206166 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-collectd-1.3.0/chefignore'
D, [2016-02-15T18:00:48.215763 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-hazelcast-1.3.0/chefignore'
D, [2016-02-15T18:00:48.223984 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-collectd-1.5.3/chefignore'
D, [2016-02-15T18:00:48.227989 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-mongodb-2.4.1/chefignore'
D, [2016-02-15T18:00:48.242163 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-collectd-1.5.2/chefignore'
D, [2016-02-15T18:00:48.246157 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/simple-mailcatcher-0.1.1/chefignore'
D, [2016-02-15T18:00:48.263806 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-kibana-storage-1.5.1/chefignore'
D, [2016-02-15T18:00:48.273278 #24690] DEBUG -- : Using 'chefignore' at '/home/sirex/.berkshelf/cookbooks/proj-rabbitmq-0.4.0/chefignore'
D, [2016-02-15T18:00:48.279886 #24690] DEBUG -- :   Detected there is a cached cookbook
D, [2016-02-15T18:00:48.280133 #24690] DEBUG -- :   Checking transitive dependencies for proj-base (0.5.1)
D, [2016-02-15T18:00:48.280174 #24690] DEBUG -- :     Checking apt (>= 0.0.0)
D, [2016-02-15T18:00:48.280247 #24690] DEBUG -- :     Checking transitive dependencies for apt (2.7.0)
D, [2016-02-15T18:00:48.280280 #24690] DEBUG -- :       Already checked - skipping
D, [2016-02-15T18:00:48.280304 #24690] DEBUG -- :     Checking chef-client (>= 0.0.0)
D, [2016-02-15T18:00:48.280346 #24690] DEBUG -- :     Checking transitive dependencies for chef-client (4.3.0)
D, [2016-02-15T18:00:48.280372 #24690] DEBUG -- :       Already checked - skipping
D, [2016-02-15T18:00:48.280409 #24690] DEBUG -- : Checking proj-chef (~> 0.1)
D, [2016-02-15T18:00:48.283470 #24690] DEBUG -- :   Detected there is a cached cookbook
D, [2016-02-15T18:00:48.283602 #24690] DEBUG -- :   Checking transitive dependencies for proj-chef (0.1.0)
D, [2016-02-15T18:00:48.283643 #24690] DEBUG -- :     Checking apt (>= 0.0.0)
D, [2016-02-15T18:00:48.283699 #24690] DEBUG -- :     Checking transitive dependencies for apt (2.7.0)
D, [2016-02-15T18:00:48.283738 #24690] DEBUG -- :       Already checked - skipping
D, [2016-02-15T18:00:48.283761 #24690] DEBUG -- :     Checking chef-client (>= 0.0.0)
D, [2016-02-15T18:00:48.283807 #24690] DEBUG -- :     Checking transitive dependencies for chef-client (4.3.0)
D, [2016-02-15T18:00:48.283835 #24690] DEBUG -- :       Already checked - skipping
D, [2016-02-15T18:00:48.283859 #24690] DEBUG -- :     Checking chef_handler (>= 0.0.0)
D, [2016-02-15T18:00:48.283911 #24690] DEBUG -- :     Checking transitive dependencies for chef_handler (1.1.6)
D, [2016-02-15T18:00:48.283940 #24690] DEBUG -- :       Already checked - skipping
D, [2016-02-15T18:00:48.283972 #24690] DEBUG -- : Checking proj-collectd (~> 1.3)
D, [2016-02-15T18:00:48.286884 #24690] DEBUG -- :   Detected there is a cached cookbook
D, [2016-02-15T18:00:48.286992 #24690] DEBUG -- :   Checking transitive dependencies for proj-collectd (1.3.0)
D, [2016-02-15T18:00:48.287023 #24690] DEBUG -- :     Checking collectd-lib (>= 0.0.0)
D, [2016-02-15T18:00:48.287084 #24690] DEBUG -- :     Checking transitive dependencies for collectd-lib (3.0.1)
D, [2016-02-15T18:00:48.287112 #24690] DEBUG -- :       Already checked - skipping
D, [2016-02-15T18:00:48.287139 #24690] DEBUG -- : Checking proj-grafana (~> 0.1)
D, [2016-02-15T18:00:48.289938 #24690] DEBUG -- :   Detected there is a cached cookbook
D, [2016-02-15T18:00:48.290054 #24690] DEBUG -- :   Checking transitive dependencies for proj-grafana (0.3.0)
D, [2016-02-15T18:00:48.290085 #24690] DEBUG -- :     Checking apt (>= 0.0.0)
D, [2016-02-15T18:00:48.290139 #24690] DEBUG -- :     Checking transitive dependencies for apt (2.7.0)
D, [2016-02-15T18:00:48.290166 #24690] DEBUG -- :       Already checked - skipping
D, [2016-02-15T18:00:48.290199 #24690] DEBUG -- :     Checking proj-collectd (>= 0.0.0)
D, [2016-02-15T18:00:48.290242 #24690] DEBUG -- :     Checking transitive dependencies for proj-collectd (1.3.0)
D, [2016-02-15T18:00:48.290269 #24690] DEBUG -- :       Already checked - skipping
D, [2016-02-15T18:00:48.290301 #24690] DEBUG -- :     Checking grafana2 (>= 0.0.0)
D, [2016-02-15T18:00:48.290346 #24690] DEBUG -- :     Checking transitive dependencies for grafana2 (0.1.5)
D, [2016-02-15T18:00:48.290382 #24690] DEBUG -- :       Already checked - skipping
D, [2016-02-15T18:00:48.290416 #24690] DEBUG -- :     Checking nginx (>= 0.0.0)
D, [2016-02-15T18:00:48.290459 #24690] DEBUG -- :     Checking transitive dependencies for nginx (2.7.6)
D, [2016-02-15T18:00:48.290486 #24690] DEBUG -- :       Already checked - skipping
D, [2016-02-15T18:00:48.290512 #24690] DEBUG -- : Checking proj-graph-explorer (~> 0.1)
D, [2016-02-15T18:00:48.293577 #24690] DEBUG -- :   Detected there is a cached cookbook
D, [2016-02-15T18:00:48.293679 #24690] DEBUG -- :   Cached cookbook has different dependencies - cannot be trusted!
The lockfile is out of sync! Run `berks install` to sync the lockfile.
        /home/sirex/.rvm/gems/ruby-2.1.5/gems/berkshelf-4.0.1/lib/berkshelf/berksfile.rb:690:in `validate_lockfile_trusted!'
        /home/sirex/.rvm/gems/ruby-2.1.5/gems/berkshelf-4.0.1/lib/berkshelf/berksfile.rb:457:in `outdated'
        /home/sirex/.rvm/gems/ruby-2.1.5/gems/berkshelf-4.0.1/lib/berkshelf/cli.rb:250:in `outdated'
        /home/sirex/.rvm/gems/ruby-2.1.5/gems/thor-0.19.1/lib/thor/command.rb:27:in `run'
        /home/sirex/.rvm/gems/ruby-2.1.5/gems/thor-0.19.1/lib/thor/invocation.rb:126:in `invoke_command'
        /home/sirex/.rvm/gems/ruby-2.1.5/gems/thor-0.19.1/lib/thor.rb:359:in `dispatch'
        /home/sirex/.rvm/gems/ruby-2.1.5/gems/berkshelf-4.0.1/lib/berkshelf/cli.rb:52:in `dispatch'
        /home/sirex/.rvm/gems/ruby-2.1.5/gems/thor-0.19.1/lib/thor/base.rb:440:in `start'
        /home/sirex/.rvm/gems/ruby-2.1.5/gems/berkshelf-4.0.1/lib/berkshelf/cli.rb:27:in `execute!'
        /home/sirex/.rvm/gems/ruby-2.1.5/gems/berkshelf-4.0.1/bin/berks:5:in `<top (required)>'
        /home/sirex/.rvm/gems/ruby-2.1.5/bin/berks:23:in `load'
        /home/sirex/.rvm/gems/ruby-2.1.5/bin/berks:23:in `<main>'
        /home/sirex/.rvm/gems/ruby-2.1.5/bin/ruby_executable_hooks:15:in `eval'
        /home/sirex/.rvm/gems/ruby-2.1.5/bin/ruby_executable_hooks:15:in `<main>'
```

**Comment by jsirex** [comment]
Berksfile:
```ruby
source 'http://berkshelf-api:4004'

group :common do
  cookbook 'proj-base', '~> 0.5'
  cookbook 'proj-chef', '~> 0.1'
end

group :storage do
  cookbook 'proj-graphite', '~> 0.3'
end

group :collector do
  cookbook 'proj-collectd', '~> 1.3'
end

group :view do
  cookbook 'proj-grafana', '~> 0.1'
  cookbook 'proj-graph-explorer', '~> 0.1'
end

group :alerting do
  cookbook 'proj-seyren', '~> 0.1'
end
```

Lock file:
```
DEPENDENCIES
  proj-base (~> 0.5)
  proj-chef (~> 0.1)
  proj-collectd (~> 1.0)
  proj-grafana (~> 0.1)
  proj-graph-explorer (~> 0.1)
  proj-graphite (~> 0.3)
  proj-seyren (~> 0.1)

GRAPH
  7-zip (1.0.2)
    windows (>= 1.2.2)
  apt (2.7.0)
  ark (0.9.0)
    7-zip (>= 0.0.0)
    windows (>= 0.0.0)
  bluepill (2.3.1)
    rsyslog (>= 0.0.0)
  build-essential (2.2.3)
  chef-client (4.3.0)
    cron (>= 1.2.0)
    logrotate (>= 1.2.0)
    windows (~> 1.11)
  chef_handler (1.1.6)
  collectd (2.1.2)
  collectd-lib (3.0.1)
  cron (1.6.1)
  proj-base (0.5.1)
    apt (>= 0.0.0)
    chef-client (>= 0.0.0)
  proj-chef (0.1.0)
    apt (>= 0.0.0)
    chef-client (>= 0.0.0)
    chef_handler (>= 0.0.0)
  proj-collectd (1.3.0)
    collectd-lib (>= 0.0.0)
  proj-grafana (0.3.0)
    apt (>= 0.0.0)
    proj-collectd (>= 0.0.0)
    grafana2 (>= 0.0.0)
    nginx (>= 0.0.0)
  proj-graph-explorer (0.2.0)
    apt (>= 0.0.0)
    proj-collectd (>= 0.0.0)
    proj-java (>= 0.0.0)
    elasticsearch (>= 0.0.0)
    nginx (>= 0.0.0)
  proj-graphite (0.3.1)
    proj-collectd (~> 1.3)
    graphite (~> 1.0)
    nginx (~> 2.7)
    runit (~> 1.6)
  proj-java (2.1.0)
    apt (>= 0.0.0)
  proj-seyren (0.3.0)
    proj-collectd (~> 1.3)
    nginx (~> 2.7)
  elasticsearch (0.3.14)
    ark (>= 0.2.4)
  grafana2 (0.1.5)
    apt (>= 0.0.0)
    yum (>= 0.0.0)
  graphite (1.0.2)
    build-essential (>= 0.0.0)
    memcached (>= 0.0.0)
    python (>= 0.0.0)
    runit (~> 1.0)
    yum-epel (>= 0.0.0)
  iptables (1.0.0)
  logrotate (1.9.2)
  memcached (1.7.2)
    runit (~> 1.0)
    yum (~> 3.0)
    yum-epel (>= 0.0.0)
  mongodb (2.1.0)
    apt (>= 0.0.0)
    build-essential (>= 0.0.0)
    iptables (>= 0.0.0)
  nginx (2.7.6)
    apt (~> 2.2)
    bluepill (~> 2.3)
    build-essential (~> 2.0)
    ohai (~> 2.0)
    runit (~> 1.2)
    yum-epel (~> 0.3)
  ohai (2.0.1)
  packagecloud (0.0.17)
  python (1.4.6)
    build-essential (>= 0.0.0)
    yum-epel (>= 0.0.0)
  rsyslog (2.1.0)
  runit (1.6.0)
    packagecloud (>= 0.0.0)
  windows (1.36.6)
    chef_handler (>= 0.0.0)
  yum (3.6.0)
  yum-epel (0.6.0)
    yum (~> 3.0)
```

**Comment by Annih** [comment]
Ah ok ... it's not the same bug :(
Yours is about a cached cookbook that diverged; you can quick fix your issue by removing `proj-graph-explorer` from your cookbook cache (e.g. `rm -rf ~/.berkshelf/cookbooks/proj-graph-explorer*`)

**Comment by jsirex** [comment]
Still having issue:
1. remove all cached cookbooks
2. remove lock file
3. do berks install
4. do berks upload - lock file out of sync :(

**Comment by bbaugher** [comment]
I think you may have run into the same as me in #1473. Deleting the cookbook from cache is the only way I know how to get around it (`berks shelf uninstall COOKBOOK`)

**Comment by trinitronx** [comment]
Frustrating, but the only way I was able to fix this was to remove all the cached cookbooks in the "shelf":

```
rm -rf ~/.berkshelf/cookbooks/
berks install
berks upload --no-freeze
```

**Comment by bbaugher** [comment]
Could try upgrading to Berkshelf 5.0 if you aren't already on that. It fixed my issue

**Comment by merqlove** [comment]
I'm found that the problem was with git repo protocols. change git:// to https://.

**Comment by mrpaws** [comment]
merqlove's solution worked for me.

**Comment by trinitronx** [comment]
Seems that I have stumbled across this bug yet again.  However, this time the old cache removal trick didn't work.  Tried it with `berkshelf` versions `5.6.5` and `6.3.1` with same result.  Not sure what could be causing it this time.  Here's a gist with my result:

https://gist.github.com/trinitronx/1350b24170b988e20a21fa9f8ff6502a

**Comment by gerard-leijdekkers** [comment]
merqlove's solution worked for me on 6.3.1.

changed
`cookbook 'wkhtmltopdf', github: 'firstbanco/chef-wkhtmltopdf'`
to
`cookbook 'wkhtmltopdf', git: 'https://github.com/firstbanco/chef-wkhtmltopdf.git'`

**Comment by lamont-granquist** [comment]
it looks like the original issue may have been fixed by #1580 

there's another issue here around the use of `github` vs `git` in the Berksfile which are distinct and will invalidate the Berksfile.lock because the compared objects are different.

ie. in the gist https://gist.github.com/trinitronx/1350b24170b988e20a21fa9f8ff6502a the "Berkshelf::GithubLocation" vs "Berkshelf::GitLocation" which are indeed different and berkshelf appears to be entirely accurately complaining that the Berksfile does not match the lockfile.

this issue seems to be compiling multiple issues/bugs which all have the same error.

if there's still an issue here -- that can be replicated on berkshelf 7.0.0 (very soon to be released) -- then it should get opened as a new issue, with a clear repo case.   i can't boil any of these comments down to a bug that still exists which needs to get fixed somewhere.
